PR: https://github.com/odoo/odoo/pull/

From: fd5361c0f7116e951cf9e69e2d97a8ccb07ce1b6
From: Christophe Matthieu
Date: 2015-07-10 15:00:15

Structural Changes: 5
Total Changes: 42

[IMP] mass_mailing: remove depends to website. Create a website_mass_mailing bridge to add subscribe snippet into website builder. Split website_links to have link_tracker to create short and trackable URLs, and website_links to have a website layout for link_tracker and add button to share page.

================================= pseudo patch: =================================

--- a/addons/mass_mailing/models/__init__.py
+++ b/addons/mass_mailing/models/__init__.py
@@ -6,4 +6,4 @@ import mail_mail
 import mail_thread
 import res_config
 import mass_mailing_report
-import website_links
+import link_tracker

--- a/addons/mass_mailing/models/website_links.py
+++ b/addons/mass_mailing/models/link_tracker.py
@@ -1,15 +1,15 @@
 from openerp import fields
 from openerp import models
 
-class website_links(models.Model):
-    _inherit = "website.links"
+class link_tracker(models.Model):
+    _inherit = "link.tracker"
 
     mass_mailing_id = fields.Many2one('mail.mass_mailing', string='Mass Mailing')
     mass_mailing_campaign_id = fields.Many2one('mail.mass_mailing.campaign', string='Mass Mailing Campaign')
 
-class website_links_click(models.Model):
-    _inherit = "website.links.click"
+class link_tracker_click(models.Model):
+    _inherit = "link.tracker.click"
 
     mail_stat_id = fields.Many2one('mail.mail.statistics', string='Mail Statistics')
     mass_mailing_id = fields.Many2one('mail.mass_mailing', string='Mass Mailing')
-    mass_mailing_campaign_id = fields.Many2one('mail.mass_mailing.campaign', string='Mass Mailing Campaign')
\ No newline at end of file
+    mass_mailing_campaign_id = fields.Many2one('mail.mass_mailing.campaign', string='Mass Mailing Campaign')

--- a/addons/mass_mailing/models/mass_mailing.py
+++ b/addons/mass_mailing/models/mass_mailing.py
@@ -208,7 +208,7 @@ class MassMailingCampaign(osv.Model):
         cr.execute("""
             SELECT COUNT(DISTINCT(stats.id)) AS nb_mails, COUNT(DISTINCT(clicks.mail_stat_id)) AS nb_clicks, stats.mass_mailing_campaign_id AS id 
             FROM mail_mail_statistics AS stats
-            LEFT OUTER JOIN website_links_click AS clicks ON clicks.mail_stat_id = stats.id
+            LEFT OUTER JOIN link_tracker_click AS clicks ON clicks.mail_stat_id = stats.id
             WHERE stats.mass_mailing_campaign_id IN %s
             GROUP BY stats.mass_mailing_campaign_id
         """, (tuple(ids), ))
@@ -436,7 +436,7 @@ class MassMailing(osv.Model):
         cr.execute("""
             SELECT COUNT(DISTINCT(stats.id)) AS nb_mails, COUNT(DISTINCT(clicks.mail_stat_id)) AS nb_clicks, stats.mass_mailing_id AS id 
             FROM mail_mail_statistics AS stats
-            LEFT OUTER JOIN website_links_click AS clicks ON clicks.mail_stat_id = stats.id
+            LEFT OUTER JOIN link_tracker_click AS clicks ON clicks.mail_stat_id = stats.id
             WHERE stats.mass_mailing_id IN %s
             GROUP BY stats.mass_mailing_id
         """, (tuple(ids), ))
@@ -583,7 +583,7 @@ class MassMailing(osv.Model):
 
     def mass_mailing_statistics_action(self, cr, uid, ids, context=None):
         res = self.pool['ir.actions.act_window'].for_xml_id(cr, uid, 'mass_mailing', 'action_view_mass_mailing_statistics', context=context)
-        link_click_ids = self.pool['website.links.click'].search(cr, uid, [('mass_mailing_id', 'in', ids)], context=context)
+        link_click_ids = self.pool['link.tracker.click'].search(cr, uid, [('mass_mailing_id', 'in', ids)], context=context)
         res['domain'] = [('id', 'in', link_click_ids)]
         return res
 
@@ -796,7 +796,7 @@ class MassMailing(osv.Model):
             if utm_mixin.medium_id:
                 vals['medium_id'] = utm_mixin.medium_id.id
 
-            res[mass_mailing.id] = self.pool['website.links'].convert_links(cr, uid, html, vals, blacklist=['/unsubscribe_from_list'], context=context)
+            res[mass_mailing.id] = self.pool['link.tracker'].convert_links(cr, uid, html, vals, blacklist=['/unsubscribe_from_list'], context=context)
 
         return res
 

--- a/addons/mass_mailing/models/mass_mailing_stats.py
+++ b/addons/mass_mailing/models/mass_mailing_stats.py
@@ -78,7 +78,7 @@ class MailMailStats(osv.Model):
         'opened': fields.datetime('Opened', help='Date when the email has been opened the first time'),
         'replied': fields.datetime('Replied', help='Date when this email has been replied for the first time.'),
         'bounced': fields.datetime('Bounced', help='Date when this email has bounced.'),
-        'links_click_ids': fields.one2many('website.links.click', 'mail_stat_id', 'Links click'),
+        'links_click_ids': fields.one2many('link.tracker.click', 'mail_stat_id', 'Links click'),
         'state': fields.function(_compute_state, string='State', type="selection", multi="state",
                                  selection=[('outgoing', 'Outgoing'),
                                             ('exception', 'Exception'),
